docs/root/configuration/listeners/network_filters/redis_proxy_filter.rst

```diff
@@ -70,3 +72,49 @@ The Redis proxy filter supports the following runtime settings:
 redis.drain_close_enabled
   % of connections that will be drain closed if the server is draining and would otherwise
   attempt a drain close. Defaults to 100.
+
+.. _config_network_filters_redis_proxy_fault_injection:
+
+Fault Injection
+---------------
+
+The Redis filter can perform fault injection. Currently, Delay and Error faults are supported.
+Delay faults delay a request, and Error faults respond with an error. Moreover, errors can be delayed.
+
+If multiple faults are specified, the fault injection percentage should not exceed 100% for a given 
+fault and Redis command combination. For example, if you have two faults; one applying to GET at 60
+%, and one applying to all commands at 50%, that is a bad configuration as GET now has 110% chance of
+applying a fault. The Redis filter does not check for correctness in your configuration- it is user's
+responsibility to make sure both the default and runtime percentages are correct!
```

Comment:
This is kind of strange. Can we just do the right checking in code and fail the config load?

Reply:
In the current scheme, we check the fault runtime value every request, if set, so we'd need to check correctness at request time if at all. A priori I expect users to change fault behavior at runtime vs at config time.

I considered only periodically checking the fault injection percentage- say every minute or so- and storing a fault table that is verified for correctness. However, it's not clear to me what Envoy behavior should be if we break fault injection at runtime- should we log an error and stop applying faults? What do you think?

Reply:
OK I see. I think this is probably fine for now, but can you potentially try to write this a bit more crisply? I found this confusing from a read through.